packages/paste-core/components/button-group/src/ButtonGroup.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {styled, css} from '@twilio-paste/styling-library';
+
+const AttachedStyled = styled.div(
+  css({
+    '& > button:focus': {
+      // NOTE: allow focus indication to remain visible on all sides of a button despite
+      // sibling elements being rendered immediately adjacent under the attached styles
+      zIndex: 'zIndex10',
+    },
+    '& > button:not(:first-of-type):not(:last-of-type)': {
+      borderRadius: 0,
+    },
+    '& > button:first-of-type': {
+      borderTopRightRadius: 0,
+      borderBottomRightRadius: 0,
+    },
+    '& > button:last-of-type': {
+      borderTopLeftRadius: 0,
+      borderBottomLeftRadius: 0,
+    },
+  })
+);
+
+const UnattachedStyled = styled.div(
+  css({
+    '& > button:not(:first-of-type)': {
+      marginLeft: 'space30',
+    },
+    '& > button:first-of-type': {
+      marginLeft: 'space0',
+    },
+  })
+);
+
+export interface ButtonGroupProps extends Omit<React.ComponentPropsWithRef<'div'>, 'children'> {
+  element?: string;
+  attached?: boolean;
+  children: React.ReactElement[];
+}
+
+export const ButtonGroup = React.forwardRef<HTMLDivElement, ButtonGroupProps>(
+  ({element = 'BUTTON_GROUP', children, attached = false, ref, ...rest}) => {
+    return (
+      <Box
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        as={attached ? AttachedStyled : UnattachedStyled}
+        ref={ref}
+        element={element}
+        display="flex"
+        role="group"
+        {...safelySpreadBoxProps(rest)}
+      >
+        {children}
```

Comment:
question: would there be any benefit to passing a `data-paste-element` prop to button group children called `BUTTON_GROUP_CHILD` or something similar? otherwise, button group buttons will inherit all custom styles applied to any buttons

Reply:
hmm..  Since this is composable and ButtonGroup doesn't export the `Button` - they're existing system components -  I suggest we leave it to developers to pass the `element` prop here for customization.